File: java/infeccion/main/java/com/ejemplo/Signer.java
package com.ejemplo;

import java.io.IOException;

public class Signer {
    public static void signApk(String apkFile, String keystore, String alias, String storepass) {
        try {
            Process process = new ProcessBuilder(
                "jarsigner",
                "-keystore", keystore,
                "-storepass", storepass,
                apkFile,
                alias
            ).inheritIO().start();
            process.waitFor();
            System.out.println("APK firmado correctamente");

            Process alignProcess = new ProcessBuilder(
                "zipalign", "-v", "4", apkFile, apkFile.replace(".apk", "_aligned.apk")
            ).inheritIO().start();
            alignProcess.waitFor();
            System.out.println("APK alineado correctamente");
        } catch (IOException | InterruptedException e) {
            System.out.println("Error al firmar el APK: " + e.getMessage());
        }
    }
}

File: python/infeccion/signer.py
import subprocess

def sign_apk(apk_file, keystore, alias, storepass):
    result = subprocess.run(['jarsigner', '-keystore', keystore, '-storepass', storepass, apk_file, alias], capture_output=True, text=True)
    if result.returncode == 0:
        print("APK firmado correctamente")
        align_result = subprocess.run(['zipalign', '-v', '4', apk_file, f"{apk_file}_aligned.apk"], capture_output=True, text=True)
        if align_result.returncode == 0:
            print("APK alineado correctamente")
        else:
            print(f"Error al alinear el APK: {align_result.stderr}")
    else:
        print(f"Error al firmar el APK: {result.stderr}")
